lib/internal/modules/cjs/loader.js

```diff
@@ -225,15 +226,26 @@ function readPackage(requestPath) {
   }
 }
 
-function tryPackage(requestPath, exts, isMain) {
-  var pkg = readPackage(requestPath);
+function tryPackage(requestPath, exts, isMain, originalPath) {
+  const pkg = readPackage(requestPath);
 
   if (!pkg) return false;
 
-  var filename = path.resolve(requestPath, pkg);
-  return tryFile(filename, isMain) ||
-         tryExtensions(filename, exts, isMain) ||
-         tryExtensions(path.resolve(filename, 'index'), exts, isMain);
+  const filename = path.resolve(requestPath, pkg);
+  const actual = tryFile(filename, isMain) ||
+    tryExtensions(filename, exts, isMain) ||
+    tryExtensions(path.resolve(filename, 'index'), exts, isMain);
+  if (actual === false) {
```

Comment:
I haven't read this closely enough, but does this also catch cases like `pkg/x.js` as well as mains?

Reply:
It catches all main entries that we can not resolve.